s3transfer/manager.py

```diff
@@ -27,26 +29,40 @@
 class TransferConfig(object):
     def __init__(self,
                  multipart_threshold=8 * MB,
-                 max_concurrency=10,
                  multipart_chunksize=8 * MB,
-                 max_queue_size=0,
+                 max_request_concurrency=10,
```

Comment:
I was not going to let them tune stuff like submission queue size and number of threads it to start. It does not make sense especially since the TransferManager is already blocking and I need to keep that backwards compatibility for that as well. So something roughly like this:
```py
from s3transfer.manager import TransferConfig as S3TransferConfig


class TransferConfig(S3TransferConfig):
    ALIAS = {
        'max_concurrency': 'max_request_concurrency'
    }
    def __init__(self,
                 multipart_threshold=8 * MB,
                 max_concurrency=10,
                 multipart_chunksize=8 * MB,
                 num_download_attempts=5,
                 max_io_queue=100):
         super(TransferConfig, self).__init__(... # Account for mappings as well)
        for alias in self.ALIAS:
            setattr(self, alias, getattr(self, self.ALIAS[alias])
```
If we want the option for the manager to be non-blocking then we would look into adding the extra tuning variables. What do you think?